src/nvim/os/pty_process.c

```diff
@@ -113,31 +114,34 @@ bool pty_process_spawn(Job *job) FUNC_ATTR_NONNULL_ALL
   int master_status_flags = fcntl(master, F_GETFL);
   if (master_status_flags == -1) {
     ELOG("Failed to get master descriptor status flags: %s", strerror(errno));
+    err = -errno;
     goto error;
   }
   if (fcntl(master, F_SETFL, master_status_flags | O_NONBLOCK) == -1) {
     ELOG("Failed to make master descriptor non-blocking: %s", strerror(errno));
+    err = -errno;
     goto error;
   }
 
   if (job->opts.writable
-      && !set_pipe_duplicating_descriptor(master, &ptyproc->proc_stdin)) {
+      && (err = set_pipe_duplicating_descriptor(master, &ptyproc->proc_stdin)))
+  {
     goto error;
   }
 
-  if (job->opts.stdout_cb
-      && !set_pipe_duplicating_descriptor(master, &ptyproc->proc_stdout)) {
+  if (job->opts.stdout_cb &&
+      (err = set_pipe_duplicating_descriptor(master, &ptyproc->proc_stdout))) {
     goto error;
   }
 
-  if (job->opts.stderr_cb
-      && !set_pipe_duplicating_descriptor(master, &ptyproc->proc_stderr)) {
+  if (job->opts.stderr_cb &&
+      (err = set_pipe_duplicating_descriptor(master, &ptyproc->proc_stderr))) {
```

Comment:
I would in this case put &ptyproc… argument on the next line, making it three lines total.